### DISCUSSION TITLE
404 for javascripts

### DISCUSSION THREAD

**Discussion by aforsythe** [squidfunk/mkdocs-material#3505] [Q&A]
I am getting a warning that there's a 404 for the javascripts

```
WARNING  -  [07:52:09] "GET /js/mathjax.js HTTP/1.1" code 404
WARNING  -  [07:52:09] "GET /js/refresh_on_toggle_dark_light.js HTTP/1.1" code 404
WARNING  -  [07:52:09] "GET /js/loader.js HTTP/1.1" code 404
WARNING  -  [07:52:09] "GET /js/mathjax.js HTTP/1.1" code 404
WARNING  -  [07:52:09] "GET /js/loader.js HTTP/1.1" code 404
WARNING  -  [07:52:09] "GET /js/refresh_on_toggle_dark_light.js HTTP/1.1" code 404
```

Matjax code renders, as does Mermaid using `mermaid2`.
The `refresh_on_toggle_dark_light.js` doesn't seem to be doing it's job as the page requires a refresh to change from dark to light diagrams.

Any reason why I should be getting these 404 errors?  Is that why my `refresh_on_toggle_dark_light` doesn't seem to work?

Using the `docker` container `squidfunk/mkdocs-material`

Any thoughts would be greatly appreciated.

**Comment by squidfunk** [comment]
Material for MkDocs doesn't provide those files, so it's probably a better idea to ask your question on the issue tracker of the project providing them, which I guess is `mkdocs-mermaid2-plugin`.

**Reply by aforsythe** [reply]
Fair enough.  
Just thought the 404 may be an issue introduced by the docker image.

**Reply by squidfunk** [reply]
If you run the Docker image with the [command mentioned in the docs](https://squidfunk.github.io/mkdocs-material/creating-your-site/#previewing-as-you-write), the assets will definitely be picked up.

**Reply by aforsythe** [reply]
Thanks for the quick reply ... yes, using the command mentioned.  Will follow up with the `mk-mermaid2-plugin` folks to see if it's an issue on their side.

**Comment by aforsythe** [comment]
Not sure where the 404 is coming from yet, but I did get `refresh_on_toggle_dark_light.js` working.  Must be in the `docs\extra` directory not just any path listed in the `extra_javascript` sections of the `mkdocs.yml`

**Reply by squidfunk** [reply]
Yes, extra CSS and JS must be located in the `docs` directory. The customization guide follows that principle:
https://squidfunk.github.io/mkdocs-material/customization/#additional-javascript
